flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorResourceUtils.java

```diff
@@ -285,4 +286,15 @@ private static <T> void setConfigOptionToDefaultIfNotSet(
             config.set(option, defaultValue);
         }
     }
+
+    private static <T> void resetConfigOption(
+            Configuration config, ConfigOption<T> option, T value) {
+        if (config.contains(option)) {
+            LOG.info(
+                    "The configuration option {} has no effect for local execution, resetting to {}",
+                    option.key(),
+                    value);
+        }
+        config.set(option, value);
+    }
```

Comment:
Then we can have a test to verify that all unused config options are being ignored.

Reply:
Yes this makes sense. After removing them, I find we can be simply set the valid default config value, instead of checking first. So `silentlySetConfigOptionIfNotSet()` and it's callsites are just removed. A bit simpler.

Reply:
@xintongsong how do I trigger a Jenkins build? I tested manually and the code snippet in the JIRA passes.

Reply:
I see you've already trigger the CI with a git push. Next time you can simply comment `@flinkbot run azure` on the PR.

Reply:
I see you've already trigger the CI with a git push. Next time you can simply comment `@flinkbot run azure` on the PR.